aptos-move/framework/aptos-framework/sources/storage_gas.move

```diff
@@ -209,32 +233,45 @@ module aptos_framework::storage_gas {
         // Check the corner case that current_usage_bps drops before the first point.
         let (left, right) = if (num_points == 0) {
             (&Point {x: 0, y: 0}, &Point {x: BASIS_POINT_DENOMINATION, y: BASIS_POINT_DENOMINATION})
+        } else if (current_usage_bps < vector::borrow(points, 0).x) {
+            (&Point { x: 0, y: 0 }, vector::borrow(points, 0))
+        } else if (vector::borrow(points, num_points-1).x <= current_usage_bps) {
+            (vector::borrow(points, num_points-1), &Point { x: BASIS_POINT_DENOMINATION, y: BASIS_POINT_DENOMINATION })
         } else {
-            let (i, j) = (0, num_points - 1);
-            while (i < j) {
+            let (i, j) = (0, num_points - 2);
```

Comment:
I understand this is trying to separate out the edge-case of the entry at index num_points - 1 but is this safe from a binary search perspective? This completely changes the cadence of the search.

cc @lightmark

Reply:
The edge cases were separated out and became upfront for the sake of loop invariant. This refactoring should preserve the semantics of the function (i.e., equivalent to the previous function), and needs a closer look from @lightmark.

Reply:
looks good to me.